host-controller/src/main/java/org/jboss/as/host/controller/HostControllerEnvironment.java

```diff
@@ -808,6 +811,15 @@ public Stability getStability() {
         return this.stability;
     }
 
+    /**
+     * Returns an unmodifiable set of all the permissible stability levels.
+     *
+     * @return a set of stability levels
+     */
+    public Set<Stability> getStabilities() {
+        return this.stabilities;
```

Comment:
... that way this method (and t can just delegate to ProductConfig.getStabilities().

I would also consider defining this method in ProcessEnvironment to enforce consistency across subclasses.

Reply:
done

Reply:
> I would also consider defining this method in ProcessEnvironment to enforce consistency across subclasses.

@pferraro @yersan This is a good idea but I think it's out of scope for this JIRA. Just because when I look at ProcessEnvironment it's an abstract class (not an interface) shares a lot of functionality with both of its subclasses, but stores and exposes no state itself and thus forces both subclasses to do so.  If we're going to change that for this we should revisit all of it, which is out of scope.

I have no idea why it's done this way. My only guess is ProcessEnvironment was originally stateless and over time the two subclasses evolved independently and the kind of refactoring I'm discussing here was never in scope. :)

I filed https://issues.redhat.com/browse/WFCORE-6760